components/json-splitter.js

```diff
@@ -0,0 +1,88 @@
+// json-splitter.js
+
+/**
+ * This component takes a json string with one or more array elements, parses it, and makes each array element into an 
+ * individual records.  Each record is adorned with a virtual node ID (vnid) to uniquely identify it in the rdf pipeline, 
+ * and sends them on to the next component.
+ */
+
+var jsonpointer = require('jsonpointer');
+
+var _ = require('underscore');
+var baseNode = require('./base-node');
+
+exports.getComponent = baseNode.updatePromise({
+
+  description: 'Splits a json dataset into individual records, adds a vnid to each, and sends them on.',
+
+  inPorts: { 
+      in: { 
+        addressable: true, 
+        datatype: 'string',
+        description: 'json string input',
+        required: true
+      },
+
+      json_pointer: {
+        addressable: true, 
+        datatype: 'string',
+        description: 'json pointer for the json data attribute to use as the vnid (RFC 6901)',
+        required: true
+      }
+  },
+
+  onchange: function( data ) {   
+
+      if ( this.in && this.json_pointer ) { 
```

Comment:
What is 'this' expected to be at this point?  An inPorts object?  But below it seems like a NoFlo component instance.  I don't understand how there is both this.in and this.outPorts .  I would think it would be this.inPorts.in .